### DISCUSSION TITLE
KB2040 support

### DISCUSSION THREAD

**Discussion by qpucka** [micropython/micropython#9846] [Hardware & Peripherals]
Will there be a KB2040 download at some point or is this something that should be configured/compiled by the user?

Thank you.

**Comment by robert-hh** [comment]
There will be one if someone makes a PR for the board files. How about you?
Just take one of the other Adafruit boards as template.

**Comment by qpucka** [comment]
@robert-hh Thank you. This is where I'm headed but wanted to make sure I'm not duplicating work, stepping on toes, etc.

I've been looking through the docs and the minimal port. Will also have a look at similar Adafruit ports as suggested.

**Reply by robert-hh** [reply]
Since it is a standard RP2040 device, you just have to create the files in a subdirectory of boards. For instance under the name ADAFRUIT_KB2040.

**Reply by scruss** [reply]
It's an RP2040 with an unusual flash configuration (8 MB WinBond) and a Neopixel. Not sure how different it is to the [SparkFun Pro Micro - RP2040](https://www.sparkfun.com/products/18288): maybe only flash size.

Note that there's no USB HID support in MicroPython, but there are a couple of open PRs that are being worked on to make it happen

**Comment by qpucka** [comment]
@robert-hh @scruss 

Thank you. PR submitted. Much less involved than I was anticipating.

**Comment by DKonneker** [comment]
I am interested in the PR for the KB2040.  Is this available?

**Reply by scruss** [reply]
it's here: [ports/rp2/boards/ADAFRUIT_KB2040: Add Adafruit KB2040 board. by qpucka · Pull Request #9859 · micropython/micropython](https://github.com/micropython/micropython/pull/9859)
